common/core/gpudevice.cpp

```diff
@@ -35,6 +35,10 @@ bool GpuDevice::Initialize() {
   if (initialization_state_ & kInitialized)
     return true;
 
+  lock_fd_ = open("/vendor/hwc.lock", O_RDONLY);
```

Comment:
lock_fd_ needs to be initialized early since HandleWait relies on this variable to determine if thread needs to fall into sleep wait or not. If HandleWait fall into sleep first time, HandleRoutine is no chance to get executed.

Reply:
hmm HandleRoutine gets called first before HandleWait, so this should be fine and that removes all need to locking in Handleoutine

Reply:
Can you actully test removing all changes done in common/core/gpudevice.cpp

Reply:
From the HWC::ProcessThread() method in common/utils/hwcthread.cpp, The HandleWait is called before the HandleRoutine().

void HWCThread::ProcessThread() {
  setpriority(PRIO_PROCESS, 0, priority_);
  prctl(PR_SET_NAME, name_.c_str());

  while (1) {
    HandleWait();
    if (exit_) {
      HandleExit();
      fd_handler_.RemoveFd(event_.get_fd());
      return;
    }

    HandleRoutine();
  }
}